text/0000-unsafe-extern-blocks.md

```diff
@@ -0,0 +1,120 @@
+
+- Feature Name: `unsafe_extern`
+- Start Date: 2023-05-23
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+In Edition 2024 it is `unsafe` to declare an `extern` function or static, but external functions and statics *can* be safe to use after the initial declaration.
+
+# Motivation
+[motivation]: #motivation
+
+Simply declaring extern items, even without ever using them, can cause Undefined Behavior.
+When performing cross-language compilation, attributes on one function declaration can flow to the foreign declaration elsewhere within LLVM and cause a miscompilation.
```

Comment:
I think I somewhat understand why this is the case in LLVM, as they have no concept of `unsafe`, and in the end the function will have only one signature, so might as well merge them and their attributes.

But maybe the RFC text could expand a little more on why this is not just a bug to fix in LLVM, or in `rustc`'s usage of LLVM, perhaps with links to open issues? Maybe an alternative fix would be to never expose `extern "C" { fn foo(); }` to LLVM if it's unused?

Reply:
In fact I think the state of whether and where exactly this is UB or whether there's a bug is currently open -- [here](https://github.com/rust-lang/rust/issues/46188#issuecomment-1829256377) and [here](https://github.com/rust-lang/unsafe-code-guidelines/issues/198#issuecomment-1857384321).

Reply:
I've incorporated a link to that first issue into the document (the second issue is linked from the first).